cmake/features.cmake

```diff
@@ -90,6 +90,11 @@ if (NOT THREADING STREQUAL "TBB" AND
     message(FATAL_ERROR "THREADING should be set to TBB (default), TBB_AUTO, OMP or SEQ")
 endif()
 
+if (APPLE AND THREADING STREQUAL "OMP")
+    message(WARNING "THREADING does not support OMP on MACOS at present, set THREADING to TBB")
+    set(THREADING "TBB")
+endif()
```

Comment:
maybe we need to remove OMP among available options for macOS here:
```
set_property(CACHE THREADING PROPERTY STRINGS "TBB" "TBB_AUTO" "OMP" "SEQ")
```
? 

And adjust the condition after it

Reply:
I had tried to remove OMP from set_property like:
`set_property(CACHE THREADING PROPERTY STRINGS "TBB" "TBB_AUTO" "SEQ")`
then set THREADING to OMP on macos, but during compilation, there was no prompt message that THREADING can not be set to OMP.
So I think set_property can not prevent OMP from being set. Am I right?

Reply:
after this line we have a condition that explicitly checks available options.

`set_property` works only for cmake-gui which shows available options visually.

Reply:
ok, I got it

Reply:
more elegant solution https://github.com/openvinotoolkit/openvino/pull/20405